File: Day03/part1.py
with open("input", "r") as f:
    counter = 0
    for line in f:
        upper_max = 0
        lower_max = 0
        for index in range(100):
            digit = int(line[index])
            if digit > upper_max:
                if index + 1 < 100:
                    upper_max = digit
                    lower_max = 0
                else:
                    lower_max = digit
            elif digit > lower_max:
                lower_max = digit
        counter += 10 * upper_max + lower_max
    print(f"The answer is: {counter}.")

File: Day03/part1.c
#include <stdio.h>
#include <stdlib.h>

int process_line(char *line) {
    int bigger_max = 0;
    int smaller_max = 0;
    for (int i = 0; i < 100; i++) {
        if (line[i] - '0' > bigger_max) {
            if (i + 1 < 100) {
                smaller_max = 0;
                bigger_max = line[i] - '0';
            }
            else {
                smaller_max = line[i] - '0';
            }
        }
        else if (line[i] - '0' > smaller_max) smaller_max = line[i] - '0';
    }
    return 10 * bigger_max + smaller_max;
}

int main(void) {
    FILE *f = fopen("input", "r");
    if (f == NULL) return 1;
    char buffer[102];
    int counter = 0;
    while (fgets(buffer, 102, f) != NULL) {
        counter += process_line(buffer);
    }
    printf("The answer is: %d.", counter);
    fclose(f);
    return 0;
}
